src/components/DateRange.js

```diff
@@ -43,16 +43,37 @@ class DateRange extends Component {
     } else {
       endDate = value;
     }
+
     // reverse dates if startDate before endDate
+    let hasBeenReversed = false;
     if (isBefore(endDate, startDate)) {
+      hasBeenReversed = true;
       [startDate, endDate] = [endDate, startDate];
     }
 
+    const inValidDatesWithinRange = disabledDates.filter(disabledDate =>
+      isWithinInterval(disabledDate, {
+        start: startDate,
+        end: endDate,
+      })
+    );
+
+    if (inValidDatesWithinRange.length > 0) {
+      if (hasBeenReversed) {
+        const minDate = new Date(Math.min.apply(null, inValidDatesWithinRange));
+        startDate = new Date(minDate.getTime() + 24 * 60 * 60 * 1000);
+      } else {
+        const maxDate = new Date(Math.max.apply(null, inValidDatesWithinRange));
+        endDate = new Date(maxDate.getTime() - 24 * 60 * 60 * 1000);
+      }
```

Comment:
It's can be cleaner:
```js
if (isStartDateSelected) {
    startDate = addDays(max(disabledDates), 1);
} else {
    endDate = addDays(min(disabledDates), -1);
}
```

Reply:
I've noticed that this method has some problems. It fails when the first selected date is inbetween two invalid dates.